urlc4364d0d12_ahmedelmorsy_cafetria_simulation_tgz-pJ8-main_MainJ8/src/simulation/global/Statistics.java

```diff
@@ -0,0 +1,370 @@
+        avgDelayInQueue = new HashMap<String, Double>();
+        maxDelayInQueue = new HashMap<String, Integer>();
+        custDelay = new HashMap<Customer, Integer>();
+    }
+
+    public static void drawQueueLen() {
+        final JFreeChart chart = ChartFactory.createXYLineChart(
+                "Line Chart Demo 6", // chart title
+                "X", // x axis label
+                "Y", // y axis label
+                new XYSeriesCollection(queueLengthDataset), // data
+                PlotOrientation.VERTICAL, true, // include legend
+                true, // tooltips
+                false // urls
+                );
+        chart.setBackgroundPaint(Color.white);
+
+        // final StandardLegend legend = (StandardLegend) chart.getLegend();
+        // legend.setDisplaySeriesShapes(true);
+
+        // get a reference to the plot for further customisation...
+        final XYPlot plot = chart.getXYPlot();
+        plot.setBackgroundPaint(Color.lightGray);
+        // plot.setAxisOffset(new Spacer(Spacer.ABSOLUTE, 5.0, 5.0, 5.0, 5.0));
+        plot.setDomainGridlinePaint(Color.white);
+        plot.setRangeGridlinePaint(Color.white);
+
+        final XYLineAndShapeRenderer renderer = new XYLineAndShapeRenderer();
+        renderer.setSeriesLinesVisible(0, false);
+        renderer.setSeriesShapesVisible(1, false);
+        plot.setRenderer(renderer);
+
+        // change the auto tick unit selection to integer units only...
+        final NumberAxis rangeAxis = (NumberAxis) plot.getRangeAxis();
+        rangeAxis.setStandardTickUnits(NumberAxis.createIntegerTickUnits());
+
+        final ChartPanel chartPanel = new ChartPanel(chart);
+        chartPanel.setPreferredSize(new java.awt.Dimension(500, 270));
+        ApplicationFrame demo = new ApplicationFrame("chart");
+        demo.setContentPane(chartPanel);
+        demo.pack();
+        RefineryUtilities.centerFrameOnScreen(demo);
+        demo.setVisible(true);
+
+    }
+
+    public static void UpdateQueueLength(int len, String queueType) {
+        ArrayList<QueueInfo> temp = queuesInfoMap.get(queueType);
+        QueueInfo info = temp.get(temp.size() - 1);
+        if (queueType == Const.HOTFOOD_SERVER) {
+            queueLengthDataset.add(info.from, info.count);
+            queueLengthDataset.add(info.from, len);
+        }
+        if (info.from == SimulationClk.clock) {
+            info.count = len;
+        } else {
+            info.to = SimulationClk.clock;
+            info = new QueueInfo();
+            info.from = SimulationClk.clock;
+            info.count = len;
+            temp.add(info);
+        }
+    }
+
+    public static void CustomersEnteredSystem(int num) {
+        QueueInfo temp = systemInfoMap.get(systemInfoMap.size() - 1);
+        temp.to = SimulationClk.clock;
+        QueueInfo info = new QueueInfo();
+        info.from = SimulationClk.clock;
+        info.count = temp.count + num;
+        systemInfoMap.add(info);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 159:164, 168:173.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.